packages/uploadthing/src/internal/node-http/getBody.ts

```diff
@@ -0,0 +1,74 @@
+import type { IncomingMessage } from "node:http";
+
+export type BodyResult =
+  | {
+      ok: true;
+      data: unknown;
+    }
+  | { ok: false; error: Error };
+export type NodeHTTPRequest = IncomingMessage & {
+  body?: unknown;
+};
+export async function getPostBody(opts: {
+  req: NodeHTTPRequest;
+  maxBodySize?: number;
+}): Promise<BodyResult> {
+  const { req, maxBodySize = Infinity } = opts;
+  return new Promise((resolve) => {
+    if ("body" in req) {
+      const isJsonType = req.headers["content-type"] === "application/json";
+
+      if (!isJsonType) {
+        resolve({
+          ok: false,
+          error: new Error("INVALID_CONTENT_TYPE"),
```

Comment:
i think we always wanna throw `UploadThingError`'s

Reply:
Fixed